recursion.rb

```diff
@@ -1,16 +1,58 @@
 
+#RC: n! = n * (n-1)!
+#BC: 1! = 1
+#BC: 0! = 1
+
+
 def fact(n)
+  if n == 1 || n == 0
+    return 1 #cant return n here bc of the second BC
+  end
+  return n * fact(n-1)
 end
 
+#BC: fib(1) = 1
+#BC: fib(2) = 1
+#fib(3) = fib(2) + fib(1)
+#fib(4) = fib(3) + fib(2)
+#fib(n) = fib(n-1) + fib(n-2)
+
 def fib(n)
+  return 0 if n == 0
+  return 1 if n == 1
+
+  fib(n - 1) + fib(n - 2)
+  
+  end
+
```

Comment:
It looks like you have an extra "end" here, which was breaking your solution when I attempted to run it.